app/segments/drill/[id]/submission/input.js

```diff
@@ -136,7 +153,114 @@ async function uploadAttempt(
   }
 }
 
-//TODO: Create a function to check leaderboard and update if needed
+//A function to check leaderboard and update if needed
+function handleLeaderboardUpdate(
+  uploadData,
+  drillInfo,
+  currentLeaderboard,
+  teamId,
+  drillId,
+  userId,
+  queryClient,
+) {
+  const leaderboardData = currentLeaderboard;
+  const invalidateKeys = [
+    ["user"],
+    ["drillInfo"],
+    ["best_attempts", teamId, drillId], // keep teamId param as it is a string argument in useLeaderboard's query key
+    ["attempts", { drillId }], // stats pages
+    ["attempts", { userId }], // for profile index (list of drill types)
+  ];
+
+  //check if the user exists on the leaderboard
+  if (leaderboardData[uploadData.uid] == undefined) {
+    console.log("User not on leaderboard, uploading this attempt");
+
+    uploadNewLeaderboard(
+      drillInfo.mainOutputAttempt,
+      uploadData.uid,
```

Comment:
just pass uploadData instead of splitting everything apart like this

Reply:
I passed uploadData into uploadNewLeaderboard function as suggested